src/NLog/Internal/ExceptionHelper.cs

```diff
@@ -78,8 +78,9 @@ public static bool IsLoggedToInternalLogger(this Exception exception)
         /// </summary>
         /// <param name="exception">The exception to check.</param>
         /// <param name="loggerContext">Target context of the exception.</param>
+        /// <param name="logFactory">For checking <see cref="LogFactory.ThrowExceptions"/> and <see cref="LogFactory.ThrowConfigExceptions"/>. Falls back to <see cref="LogManager"/> </param>
         /// <returns><c>true</c>if the <paramref name="exception"/> must be rethrown, <c>false</c> otherwise.</returns>
-        public static bool MustBeRethrown(this Exception exception, IInternalLoggerContext loggerContext = null)
+        public static bool MustBeRethrown(this Exception exception, IInternalLoggerContext loggerContext = null, LogFactory logFactory = null)
```

Comment:
Maybe add LoggingConfiguration as property to `IInternalLoggerContext`, and then extract the LogFactory from that?

Reply:
OK will do!